test/Integration/SqlTriggerBindingIntegrationTests.cs

```diff
@@ -654,7 +654,91 @@ public async void LastAccessTimeColumn_Created_OnStartup()
+                {
+                    string json = e.Data[(index + messagePrefix.Length)..];
+                    // Sometimes we'll get messages that have extra logging content on the same line - so to prevent that from breaking
+                    // the deserialization we look for the end of the changes array and only use that.
+                    // (This is fine since we control what content is in the array so know that none of the items have a ] in them)
+                    json = json[..(json.IndexOf(']') + 1)];
+                    IReadOnlyList<SqlChange<ProductColumnTypes>> changes;
+                    try
+                    {
+                        changes = Utils.JsonDeserializeObject<IReadOnlyList<SqlChange<ProductColumnTypes>>>(json);
+                    }
+                    catch (Exception ex)
+                    {
+                        throw new InvalidOperationException($"Exception deserializing JSON content. Error={ex.Message} Json=\"{json}\"", ex);
+                    }
+                    Assert.Equal(SqlChangeOperation.Insert, changes[0].Operation); // Expected change operation
+                    ProductColumnTypes product = changes[0].Item;
+                    Assert.NotNull(product); // Product deserialized correctly
+                    Assert.Equal(expectedResponse, product); // The product has the expected values
+                    taskCompletion.SetResult(true);
+                }
+            };
+
+            // Set up listener for the changes coming in
+            foreach (Process functionHost in this.FunctionHostList)
+            {
+                functionHost.OutputDataReceived += MonitorOutputData;
+            }
+
+            // Now that we've set up our listener trigger the actions to monitor
+            string datetime = "2022-10-20 12:39:13.123";
+            this.ExecuteNonQuery("INSERT INTO [dbo].[ProductsColumnTypes] VALUES (" +
+                "999, " + // ProductId,
+                "999, " + // BigInt
+                "1, " + // Bit
+                "1.2345, " + // DecimalType
+                "1.2345, " + // Money
+                "1.2345, " + // Numeric
+                "1, " + // SmallInt
+                "1.2345, " + // SmallMoney
+                "1, " + // TinyInt
+                ".1, " + // FloatType
+                ".1, " + // Real
+                $"CONVERT(DATE, '{datetime}'), " + // Date
+                $"CONVERT(DATETIME, '{datetime}'), " + // Datetime
+                $"CONVERT(DATETIME2, '{datetime}'), " + // Datetime2
+                $"CONVERT(DATETIMEOFFSET, '{datetime}'), " + // DatetimeOffset
+                $"CONVERT(SMALLDATETIME, '{datetime}'), " + // SmallDatetime
+                $"CONVERT(TIME, '{datetime}'), " + // Time
+                "'test', " + // CharType
+                "'test', " + // Varchar
+                "'test', " + // Nchar
```

Comment:
When I used NCHAR(0xD84C) here instead of 'test' (which is what we do in the [input/output binding tests](https://github.com/Azure/azure-functions-sql-extension/blob/ff2c3096305c6d41be969b6b75c17a5c57e345a7/test/Integration/SqlInputBindingIntegrationTests.cs#L168C29-L168C29)) the test failed in the pipeline but passed locally on my machine. Will try to keep investigating why it was failing in the pipelines but using 'test' for now so I can get the test in first.